src/core/device.c

```diff
@@ -305,6 +305,19 @@ static int device_coldplug(Unit *u) {
                 found &= ~DEVICE_FOUND_UDEV;
                 if (state == DEVICE_PLUGGED)
                         state = DEVICE_TENTATIVE;
+
+                /* Also check the existence of the device syspath. Otherwise, if the device has been removed
+                 * during the switching root, it will never go to inactive state, as both Device.found and
+                 * Device.enumerated_found do not have the DEVICE_FOUND_UDEV flag, so device_catchup() in
+                 * device_update_found_one() does nothing in most cases. See issue #25106. Note, the syspath
+                 * field only serialized and deserialized when the device was previously processed by udevd,
+                 * and switched from the root filesystem with newer systemd. */
```

Comment:
```
                /* Also check the validity of the device syspath. Without this check, if the device was removed
                 * while switching root, it would never go to inactive state, as both Device.found and
                 * Device.enumerated_found do not have the DEVICE_FOUND_UDEV flag, so device_catchup() in
                 * device_update_found_one() does nothing in most cases. See issue #25106. Note that the syspath
                 * field is only serialized when systemd is sufficiently new and the device has been already processed
                 * by udevd. */
```